### DISCUSSION TITLE
iCloudWarning (WARN): Expected X CPLMaster & X CPLAsset records

### DISCUSSION THREAD

**Discussion by JyFu-py** [steilerDev/icloud-photos-sync#153] [Q&A]
Hello,

when I see this errors
![image](https://user-images.githubusercontent.com/67822884/211214976-76fd0aba-6da8-442a-b7c1-62549e4521e2.png)

you say
> **Expected <n> CPLMaster & <m> CPLAsset records, but got <x> CPLMaster & <y> CPLAsset records for album <someAlbum>**
> In an initial set, the API is queried for the number of assets in a given album. This is followed by a query to get all assets associated with the album. For some reason those APIs sometimes return contradicting results. (I suspect an error within the iCloud backend)
https://steilerdev.github.io/icloud-photos-sync/user-guides/common-warnings/

But should we do or correct something and care about these messages or just ignore them?

Thanks !

**Comment by steilerDev** [comment]
I actually don't know to be honest.

What happens is the following:

1. I'm asking iCloud for a count on the `All Photos` folder
2. I'm asking iCloud for a list of albums and folders + ask for a count on each album
3. For each album (that includes `All Photos`) I'm then asking for all assets' metadata, sorted by date, in order to diff and download

This warning is produced, if the first to queries returns a different count, than the number of assets in the third query - I strongly believe this might either be a bug in the way the first two queries are generating the count, or Apple thinks they got the asset, but once actually looking for it they can't find it.

So I'm not sure what to do about it.
